src/structures/InteractionCollector.js

```diff
@@ -33,21 +33,28 @@ class InteractionCollector extends Collector {
 
     /**
      * The message from which to collect interactions, if provided
-     * @type {?Message}
+     * @type {?Snowflake}
      */
-    this.message = options.message ?? null;
+    this.messageId = options.message?.id ?? options.message ?? null;
 
     /**
      * The channel from which to collect interactions, if provided
-     * @type {?(TextChannel|DMChannel|NewsChannel)}
+     * @type {?Snowflake}
      */
-    this.channel = this.message?.channel ?? options.channel ?? null;
+    this.channelId =
+      options.message?.channel?.id ?? options.message?.channel_id ?? options.channel?.id ?? options.channel ?? null;
```

Comment:
same here, perhaps using `this.client.channels.resolveId` but would require some modifications to accept raw api data so unsure

Reply:
It's quite horribly chained I agree, using `resolveId` will help a bit I suppose.
```js
this.channelId = this.client.channels.resolveId(options.message?.channel) ?? options.message.channel_id ?? this.client.channels.resolveId(options.channel) ?? null;
```